src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/GEOS_LandAssimGridComp.F90

```diff
@@ -1277,13 +1320,11 @@ subroutine Initialize(gc, import, export, clock, rc)
     call MPI_BCAST(N_obsbias_max,         1, MPI_INTEGER,        0,MPICOMM,mpierr)
    
 
-    ! create LandAssimAlarm with option of centered update 
+    ! create LandAssimAlarm
 
-    if (centered_update)then
-       rrTime = CurrentTime+Landassim_DT/2-ModelTimeStep 
-    else
-       rrTime = CurrentTime+Landassim_DT-ModelTimeStep 
-    endif
+    ! RR20210506: If AssimTime is indeed the time of the first land analysis, this should be "rrTime=AssimTime-ModelTimeStep"
```

Comment:
We don't do the analysis at the very beginning. Do we?

Reply:
it has not  been done in ldas.

Reply:
No, there is no analysis at the restart time.  This may be useful to have in the future.  For now, however, we can assume that the first analysis has to be after the first model_dtstep but no later than after 24 hours.

Reply:
Before I do further changes, I want to make sure what ref_time means. For example, if the program starts at 6z , and does analysis at 7:30, according to your definition, ref_time = 1:30, according to mine, it is 7:30.  We only need one definition. @gmao-rreichle

Reply:
The ref_time cannot depend on the restart time.  It has to be more general.  Two variables, ref_time and landassim_dt, determine an infinite sequence of land analysis times (see my comments in the code).  In the example above, the user must be able to specify either ref_time=1:30 **or** ref_time=7:30 **or** ref_time=22:30, and the effect must always be that a land analysis happens whenever the clock hits 1:30, 4:30, 7:30, 10:30, ..., 22:30 on any day.   Otherwise, the user would have to change the ref_time whenever they restart the program from a different time-of-day.
I suppose a modulus operation on the ref_time at initialization ( mod(ref_time_in_seconds/dtstep_in_seconds) ) could be used to have the ref_time always be the earliest possible time-of-day at which a land analysis should occur for the purpose of standardizing calculations within the code.  Again, though, specifying 0<=ref_time+N*landassim_dt<=23:59:59 for any N (within the 0...23:59:59 limits) must results in the same outcome.

Reply:
@gmao-rreichle , The ref_time defined above makes things a little bit too complicated.  If we defined ref_time as a number between 0 and assim_dtstep and the assim_time = currentTime + ref_time + n * assime_dtstep ( n=0,1,2,......) , that would be easier.